core/src/protocol.ts

```diff
@@ -1037,19 +1034,24 @@ export class ProtocolHandler implements Dispatcher<ParserEvent> {
     }
     this.muxSubscriptions.close();
     this.subscriptions.close();
-    this.listeners.forEach((l) => {
-      l.stop();
-    });
+    const proms = [];
+    for (let i = 0; i < this.listeners.length; i++) {
+      const qi = this.listeners[i];
+      if (qi) {
+        qi.stop();
+        proms.push(qi.iterClosed);
+      }
+    }
+    if (proms.length) {
+      await Promise.all(proms);
```

Comment:
what happens if there was multiple proms and one threw an error?

Reply:
those don't throw - the could resolve to an error but they don't reject